frontend/packages/console-app/src/components/nodes/NodeLogsUnitFilter.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { Chip, ChipGroup, TextInput } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { getQueryArgument, setQueryArgument } from '@console/internal/components/utils';
+
+type NodeLogsUnitFilterProps = {
+  onChangeUnit: (value: string) => void;
+  unit: string;
+};
+
+const NodeLogsUnitFilter: React.FC<NodeLogsUnitFilterProps> = ({ onChangeUnit }) => {
+  const unitQueryArgument = 'unit';
+  const unitQueryParam = getQueryArgument(unitQueryArgument);
+
+  const [value, setValue] = React.useState('');
+  const [values, setValues] = React.useState(
+    unitQueryParam?.length > 0 ? unitQueryParam.split(',') : [],
+  );
+  const { t } = useTranslation();
+
+  const getValuesString = (valuesString: string[]) => valuesString.join(',');
+
+  React.useEffect(() => {
+    const listener = (event) => {
+      if ((event.code === 'Enter' || event.code === 'NumpadEnter') && value !== '') {
+        event.preventDefault();
+        values.push(value);
+        onChangeUnit(getValuesString(values));
+        setValue('');
+      }
+    };
+    document.addEventListener('keydown', listener);
+    return () => {
+      document.removeEventListener('keydown', listener);
+    };
+  }, [onChangeUnit, value, values]);
+
+  const onChange = (newValue: string) => setValue(newValue);
+  const deleteValue = (id: string) => {
+    const copyOfValues = values;
```

Comment:
This isn't really a copy. Do you mean...?

```suggestion
    const copyOfValues = [...values];
```